common/sgx/tcbinfo.c

```diff
@@ -451,19 +558,66 @@ static oe_result_t _read_tcb_info(
         }
     }
 
-    OE_TRACE_VERBOSE("Reading tcbLevels");
-    OE_CHECK(_read_property_name_and_colon("tcbLevels", itr, end));
-    OE_CHECK(_read('[', itr, end));
-    while (*itr < end)
+    if (parsed_info->version == 2)
     {
-        OE_CHECK(_read_tcb_level(itr, end, platform_tcb_level, parsed_info));
-        // Read end of array or comma separator.
-        if (*itr < end && **itr == ']')
-            break;
+        OE_TRACE_VERBOSE("V2: Reading tcbType");
+        OE_CHECK(_read_property_name_and_colon("tcbType", itr, end));
+        OE_CHECK(_read_integer(itr, end, &value));
+        parsed_info->tcb_type = (uint32_t)value;
+        OE_CHECK(_read(',', itr, end));
 
+        OE_TRACE_VERBOSE("V2: Reading tcbEvaluationDataNumber");
+        OE_CHECK(
+            _read_property_name_and_colon("tcbEvaluationDataNumber", itr, end));
+        OE_CHECK(_read_integer(itr, end, &value));
+        parsed_info->tcb_evaluation_data_number = (uint32_t)value;
         OE_CHECK(_read(',', itr, end));
+
+        OE_TRACE_VERBOSE("Reading tcbLevels (V2)");
+        OE_CHECK(_read_property_name_and_colon("tcbLevels", itr, end));
+        OE_CHECK(_read('[', itr, end));
+        while (*itr < end)
+        {
+            OE_CHECK(_read_tcb_info_tcb_level_v2(
+                tcb_info_json,
+                itr,
+                end,
+                platform_tcb_level,
+                &parsed_info->tcb_level));
+
+            // Optimization
+            if (platform_tcb_level->status.AsUINT32 !=
+                OE_TCB_LEVEL_STATUS_UNKNOWN)
+            {
+                // Found matching TCB level, go to the end of the array.
+                while (*itr < end && **itr != ']')
+                    (*itr)++;
+            }
+
+            // Read end of array or comma separator.
+            if (*itr < end && **itr == ']')
+                break;
+
+            OE_CHECK(_read(',', itr, end));
+        }
+        OE_CHECK(_read(']', itr, end));
+    }
+    else
```

Comment:
>else [](start = 4, length = 4)

I'd prefer "else if (version == 1)" to be specific. This prevents error while parsing version 3 and above.

Reply:
Good idea, I also added code to explicitly issue an error for unsupported versions.